nexus_constructor/add_component_window.py

```diff
@@ -627,3 +651,27 @@ def update_pixel_input_validity(self):
 
 def get_fields_and_update_functions_for_component(component: Component):
     return get_fields_with_update_functions(component)
+
+
+def add_fields_to_component(
+    component: Group, fields_widget: QListWidget, component_model: NexusTreeModel = None
+):
+    """
+    Adds fields from a list widget to a component.
+    :param component: Component to add the field to.
+    :param fields_widget: The field list widget to extract field information such the name and value of each field.
+    """
+    for i in range(fields_widget.count()):
+        widget = fields_widget.itemWidget(fields_widget.item(i))
+        try:
+            component[widget.name] = widget.value
+            if component_model:
+                component_model.rowCount(component_model.current_nxs_obj[1])
```

Comment:
Do we need this in every iteration of `for` loop?

Reply:
Nope, just moved it out of the for loop. Just pushed the changed.